### ISSUE TITLE
`_process` should be removed from `BaseHandler`

### ISSUE THREAD

**Issue by NucciTheBoss** [NucciTheBoss/cleantest#31]
Process should be removed from `BaseHandler` because, quite frankly, it adds nothing of real value to the parent class:

https://github.com/NucciTheBoss/cleantest/blob/ed400422496ca0d2db8edc3623b6c4f71adb6e50/src/cleantest/provider/_handler/lxd_handler.py#L133-L142

All this function does is take in raw result data and massage it into type `Result`. However, looking at the function body above, the massaging is incredibly simple. As such, this method should be removed and not made required for handler classes. The raw result data should be massaged by the `_execute` method.

TL;DR - Refactor this method out of `BaseHandler` and `LXDHandler`.
